### PR TITLE
PR to test #246

### PR DESCRIPTION
**Pull request by mokagio**

### CODE REVIEW

File: Dangerfile [hunk 1]
```diff
@@ -22,3 +22,21 @@ Please run `bundle install` to make sure they match.
 }
   fail(message)
 end
+
+warn %{
```
[the remark this answers is not in the document]
Comment (AliSoftware) [reply Dangerfile]: Ok rubocop, but then why didn't you warn me about this exact same usage of `%{…}` on lines 18-22 too then?

**Comment by AliSoftware** [comment]
@mokagio Reopened this PR to test the "renaming file" case, and it seems that indeed it doesn't detect the new file.

Hacked the `Dangerfile` to print some debug info about the various `git.*_files` lists, see [this comment](https://github.com/wordpress-mobile/release-toolkit/pull/247#issuecomment-831972829). As you can see, while I renamed the file `ci_helper_spec.rb` to `circleci_helper_spec.rb` in the previous commit:
 - the `modified_files` array contains `spec/ci_helper_spec.rb` – so the old name
 - the `renamed_files` then indicate the old => new name

And that explains why the lint doesn't warn about any violation that we expected it to find in `spec/circleci_helper_spec.rb`, since it didn't lint that file but tried to lint the old name instead.

**Comment by AliSoftware** [comment]
A bit unsure why Danger and Rubocop properly want to [report more violations](https://github.com/wordpress-mobile/release-toolkit/pull/247/checks?check_run_id=2508816748) (11 errors total, see below), but the GitHub Action only generates 3 GitHub inline comments instead of the whole list of 11 that the Danger log emits.
Is it a limitation of GitHub Actions, or a limitation or bug of Danger? 🤔 

PS: tried to check, on a [brand new PR](https://github.com/wordpress-mobile/release-toolkit/pull/251), that this behavior wasn't because of the history on this PR (i.e Danger already having commented on it in the past and trying to update comments, as opposed to creating new ones on fresh PR). No dice.

```
Errors:
- [ ] spec/circleci_helper_spec.rb#L4: Do not use multiple top level describes - try to nest them.
- [ ] spec/circleci_helper_spec.rb#L4: Repeated describe block description on line(s) [18]
- [ ] spec/circleci_helper_spec.rb#L18: Repeated describe block description on line(s) [4]
- [ ] spec/circleci_helper_spec.rb#L20: Use `described_class` instead of `Fastlane::Helper::CircleCIHelper`.
- [ ] spec/circleci_helper_spec.rb#L23: Start context description with 'when', 'with', or 'without'.
- [ ] spec/circleci_helper_spec.rb#L24: Use `described_class` instead of `Fastlane::Helper::CircleCIHelper`.
- [ ] spec/circleci_helper_spec.rb#L28: Use `described_class` instead of `Fastlane::Helper::CircleCIHelper`.
- [ ] spec/circleci_helper_spec.rb#L33: Start context description with 'when', 'with', or 'without'.
- [ ] spec/circleci_helper_spec.rb#L34: Use `described_class` instead of `Fastlane::Helper::CircleCIHelper`.
- [ ] spec/circleci_helper_spec.rb#L35: Example has too many lines [6/5].
- [ ] spec/circleci_helper_spec.rb#L40: Name your test subject if you need to reference it explicitly.
```
image

(Pinging @orta in hope he can shed some light on this, with his knowledge of both Danger and of MS/GitHub Actions 🤔 )

File: Dangerfile [hunk 2]
```diff
@@ -22,3 +22,12 @@ Please run `bundle install` to make sure they match.
 }
   fail(message)
 end
+
+# Lint with Rubocop and report violations inline in GitHub
+github.dismiss_out_of_range_messages # This way, fixed violations should go
+renaming_map = (git.renamed_files || []).map { |e| [e[:before], e[:after]] }.to_h # when files are renamed, git.modified_files contains old name, not new one, so we need to do the convertion
```
Comment (AliSoftware) [inline Dangerfile:28]: Also submitted a similar fix on https://github.com/ashfurrow/danger-rubocop/pull/50

File: Dangerfile [hunk 3]
```diff
@@ -22,3 +22,12 @@ Please run `bundle install` to make sure they match.
...
+rubocop.lint(
+  files: git.added_files + (git.modified_files.map { |f| renaming_map[f] || f }) - git.deleted_files,
```
Comment (AliSoftware) [inline Dangerfile:30]: Also submitted a similar fix on https://github.com/ashfurrow/danger-rubocop/pull/50

**Comment by orta** [comment]
Maybe there are GitHub API issues with submitting a bunch of code review comments on similar lines but I have no idea, I don't use the file options in Danger

**Comment by AliSoftware** [comment]
> Maybe there are GitHub API issues with submitting a bunch of code review comments on similar lines but I have no idea, I don't use the file options in Danger

 Hey Orta 👋  Thanks for taking a look anyway!

Yeah I can confirm that if we don't use the `inline_comment` mode of the plugin, and instead make Danger generate a single GitHub comment (with the table of all the violations at once) we get the full list of violations at least.
Which definitively leads me to believe it is more likely to be related to a GitHub/GHA/API limitation rather than Danger [herself](https://danger.systems/guides/faq.html#danger-keeps-getting-referred-to-as-quot-her-she-quot-what-gives).

**Comment by AliSoftware** [comment]
Ah, just saw https://github.com/danger/danger-js/issues/1134 \cc @mokagio so probably related. At least it's not something specific to us 😉

**Pull request closed without merging**